### PR TITLE
GT-860 update TabsController to utilize data binding

### PR DESCRIPTION
**Pull request by frett**
Continuation of the refactoring of the tract renderer to prepare for rendering training tips.

The overall design change for the renderer is to move as much of the view layout & data binding into the view layout XML.

Once this refactoring is done I will be utilizing dagger to inject dependencies into the UI controller hierarchy (specifically EventBus) and set a flag at the root of the UI controller hierarchy specifying if tips should be rendered or not

Changes include:
- Switch from ButterKnife to Data Binding
- Convert code to Kotlin
- Only generate view hierarchy for active tab to reduce memory and CPU usage on UI thread

### CODE REVIEW

**Comment by wrandall22** [comment]
It might be nice to have a PR description to explain why this change is being made.

File: ui/tract-renderer/src/main/java/org/cru/godtools/tract/ui/controller/TabsController.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    @CallSuper
+    override fun onContentEvent(event: Event) {
+        super.onContentEvent(event)
+        checkForTabEvent(event)
+        propagateEventToChildren(event)
+    }
+
+    override fun onTabSelected(tab: TabLayout.Tab) {
+        val holder = showTabContent(tab.position) ?: return
+        if (!bindingTabs) holder.trackSelectedAnalyticsEvents()
```
Comment (wrandall22) [inline ui/tract-renderer/src/main/java/org/cru/godtools/tract/ui/controller/TabsController.kt]: Curious on style here. I know for Java, we made the [decision](https://docs.google.com/document/d/16TGhx6nL3d3cjN8Rmpo-oY55ErTSUT6sZx8RVDTpgJU/edit) (department-wide) not to inline conditional statements. Is there a reason to inline them for Kotlin?

**Comment by wrandall22** [comment]
> Only generate view hierarchy for active tab to reduce memory and CPU usage on UI thread

So SPA of you :)

File: ui/tract-renderer/src/main/res/layout/tract_content_tabs.xml [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
-        <com.google.android.material.tabs.TabLayout
+        <androidx.appcompat.widget.AppCompatTabLayout
             android:id="@+id/tabs"
             style="@style/Widget.GodTools.Tract.Content.Tabs"
             android:layout_height="wrap_content"
-            android:layout_centerHorizontal="true" />
+            android:layout_centerHorizontal="true"
+            app:backgroundTintCompat="@{StylesKt.getPrimaryColor(BaseModelKt.getStylesParent(model))}"
+            app:tabSelectedTextColor="@{StylesKt.getPrimaryTextColor(BaseModelKt.getStylesParent(model))}"
+            app:tabTextColor="@{StylesKt.getPrimaryColor(BaseModelKt.getStylesParent(model))}" />
```
[the remark this answers is not in the document]
Comment (frett) [reply ui/tract-renderer/src/main/res/layout/tract_content_tabs.xml:25]: the background is rounded corner box outline, so `backgroundTint` is tinting the outline with the correct color. the `tabTextColor` is displayed on top of a transparent part of the background and matches the outline color.

Screen Shot 2020-07-10 at 11 53 15 AM
example from our renderer test stage tool, the blue outline is `backgroundTint`, and the "2" text is `tabTextColor`

[the remark this answers is not in the document]
Comment (wrandall22) [reply ui/tract-renderer/src/main/res/layout/tract_content_tabs.xml:25]: Ah, makes sense.

File: ui/tract-renderer/src/main/java/org/cru/godtools/tract/ui/controller/TabsController.kt (hunk 1 above)
Comment (frett) [reply ui/tract-renderer/src/main/java/org/cru/godtools/tract/ui/controller/TabsController.kt]: either syntax is valid, and there are not any official Kotlin styles one way or the other that I'm aware of

I've been preferring one line if statements when it's simple logic to help consolidate the code.

File: ui/tract-renderer/src/main/res/layout/tract_content_tabs.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (frett) [reply ui/tract-renderer/src/main/res/layout/tract_content_tabs.xml:25]: also, I love being able to use data binding to express what the color should be right where I'm defining my view layout. So much nicer than customizing colors in a random Java/Kotlin file separated from the layout itself.

File: ui/tract-renderer/src/main/java/org/cru/godtools/tract/ui/controller/TabsController.kt (hunk 1 above)
Comment (frett) [reply ui/tract-renderer/src/main/java/org/cru/godtools/tract/ui/controller/TabsController.kt]: I think at the core I'm viewing Kotlin as being more similar to script languages than Java, and script languages tend to do one line if's vs blocks

**Pull request merged**